qiskit/transpiler/passes/mapping/enlarge_with_ancilla.py

```diff
@@ -43,16 +43,18 @@ def run(self, dag):
         Raises:
             TranspilerError: If there is not layout in the property set or not set at init time.
         """
-        if self.layout is None:
-            if self.property_set["layout"]:
-                self.layout = self.property_set["layout"]
-            else:
-                raise TranspilerError(
-                    "EnlargeWithAncilla requires self.property_set[\"layout\"] to run")
+        if self.property_set["layout"] is not None:
+            layout = self.property_set["layout"]
+
+        if self.layout is not None:
+            layout = self.layout
+
+        if layout is None:
+            raise TranspilerError("EnlargeWithAncilla requires property_set[\"layout\"] to run")
```

Comment:
This message is not strictly true, since this pass can also be run by just initializing it with a layout.

Reply:
I'm not fully sure what you mean. `EnlargeWithAncilla` modifies a layout. So it requires a layout to modify.

Reply:
I'm saying your message makes it sound like `property_set['layout']` must be set. However, this pass also works if the Layout is an init argument to it. This is the same discussion about where Layout should be. I think it should definitely NOT be in two places.

Reply:
Message fixed in ffae6398